src/components/LineChart/index.js

```diff
@@ -111,84 +116,116 @@ class LineChartComponent extends Component {
     return xAxisHeight + (contextChart.height || 100);
   };
 
-  getYAxisCollectionWidth = () => {
-    const { collections, series, yAxisDisplayMode, yAxisWidth } = this.props;
-    const counts = {};
-    const addCount = item => {
-      if (item.hidden) {
-        return;
-      }
-      if (item.collectionId) {
-        return;
-      }
-      const mode = (item.yAxisDisplayMode || yAxisDisplayMode).id;
-      counts[mode] = (counts[mode] || 0) + 1;
-    };
-    series.forEach(addCount);
-    collections.forEach(addCount);
-    const w1 = AxisDisplayMode.ALL.width(
-      yAxisWidth,
-      counts[AxisDisplayMode.ALL.id] || 0
-    );
-    const w2 = AxisDisplayMode.COLLAPSED.width(
+  getYAxisCollectionWidth = placement => {
+    const {
+      collections,
+      series,
+      yAxisDisplayMode,
+      yAxisPlacement,
       yAxisWidth,
-      counts[AxisDisplayMode.COLLAPSED.id] || 0
-    );
-    const width = w1 + w2;
-    return width;
+    } = this.props;
+
+    const displayModeFilter = mode => item =>
+      (item.yAxisDisplayMode || yAxisDisplayMode) === mode;
+
+    const filteredItems = []
+      .concat(series)
+      .concat(collections)
+      .filter(item => !item.hidden)
+      .filter(item => item.collectionId === undefined)
+      .filter(
+        item =>
+          (item.yAxisPlacement || yAxisPlacement) &&
+          ((item.yAxisPlacement || yAxisPlacement) === AxisPlacement.BOTH ||
+            (item.yAxisPlacement || yAxisPlacement) === placement)
+      );
+
+    const hasCollapsed =
+      filteredItems.filter(displayModeFilter(AxisDisplayMode.COLLAPSED))
+        .length > 0;
+
+    return filteredItems
+      .filter(displayModeFilter(AxisDisplayMode.ALL))
+      .reduce((acc, item) => {
+        // COLLAPSED items are already accounted-for with the initial value.
+        if (item.yAxisDisplayMode === AxisDisplayMode.COLLAPSED) {
+          return acc;
+        }
+        return acc + yAxisWidth;
+      }, hasCollapsed ? yAxisWidth : 0);
+  };
+
+  getYAxisPlacement = () => {
+    const { collections, series, yAxisPlacement } = this.props;
+    const yAxisPlacements = []
+      .concat(series.filter(s => s.collectionId === undefined))
+      .concat(collections)
+      .reduce(
+        (acc, item) => {
+          if (item.yAxisPlacement) {
+            acc[item.yAxisPlacement] = (acc[item.yAxisPlacement] || 0) + 1;
+          }
+          return acc;
+        },
+        { [yAxisPlacement]: 1 }
```

Comment:
I don't understand the initial value here, I'm sure it's on my end though.

Reply:
It's to account for the default. Example:
```
<LineChart
  yAxisPlacement={AxisPlacement.BOTH}
  series={[
    {id: 1, yAxisPlacement: AxisPlacement.LEFT},
    {id: 2},
  ]}
/>
```

In this case, you'll have `2` on both axes, and `1` only on the left. So you want the output of this reducer to be:
```
{
  BOTH: 1,
  LEFT: 1,
}
```

Does that clarify things? I'll put in a comment to explain the gist of this in the code.

Reply:
Actually, scratch that. In the process of adding a comment, I realized that it's not actually totally correct -- it would double-count the default.

I reworked it and added a story to verify the behavior -- please take a look!